### DISCUSSION TITLE
question/promise-then-callbacks

### DISCUSSION THREAD

**Discussion by giscus** [us-fe/us-fe.github.io#48] [Announcements]
# question/promise-then-callbacks

北美前端面试攻略

https://us-fe.github.io/question/promise-then-callbacks.html

**Comment by xiaoyu2er** [comment]
答案是 6，知识点 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

> onFulfilled Optional: A Function called if the Promise is fulfilled. This function has one argument, the fulfillment value. If it is not a function, it is internally replaced with an "Identity" function (it returns the received argument).

也就是说 onFulfilled 必须是个函数否则就用 x => x 来替代


```js
Promise.resolve(1) // 1
.then(() => 2) // 2
.then(3) // 替换成 x=>x // 2
.then((value) => value * 3) // 6
.then(Promise.resolve(4)) // 替换成 x => x  // 6
.then(console.log) // 6
```
